dev-tools/mage/gotest.go

```diff
@@ -273,9 +274,15 @@ func GoTest(ctx context.Context, params GoTestArgs) error {
 
 	var testArgs []string
 
-	// -race is only supported on */amd64
-	if os.Getenv("DEV_ARCH") == "amd64" {
-		if params.Race {
+	if params.Race {
+		// Enable the race detector for supported platforms.
+		// This is an intersection of the supported platforms for beats and Go.
+		//
+		// See https://go.dev/doc/articles/race_detector#Requirements.
+		raceAmd64 := os.Getenv("DEV_ARCH") == "amd64"
+		raceArm64 := os.Getenv("DEV_ARCH") == "arm64" &&
+			slices.Contains([]string{"linux", "darwin"}, os.Getenv("DEV_OS"))
+		if raceAmd64 || raceArm64 {
 			testArgs = append(testArgs, "-race")
```

Comment:
Previously this was failing silently. If RACE_DETECTOR=1 was supplied but the test was running on arm64 it would silently run the tests without race. I believe we were not even testing race on arm64 platforms until now.

Should I update this to fail if RACE_DETECTOR=1 is provided but the platform is not supported? For example for a future windows/arm64 port?

Reply:
I don't have a strong preference on this.

But I think if you request the race detector, but the platform doesn't support it we still want the tests to run.  My reasoning is that you can specify multiple platforms and race detector and it will do the best it can.  If you don't support this, then you have to specifically run the tests twice, once for platforms that support race detector and once for ones that don't.

I would like an error message that the race detector isn't supported on the specified platform, but continuing anyway.

Reply:
Thanks Lee, I updated the code to print a log message as you suggested.